### ISSUE TITLE
Barchart, 3 columns (x-values), 2 properties (y-values) per column

### ISSUE THREAD

**Issue by lfdversluis** [PhilJay/MPAndroidChart#613]
Hi, maybe a silly question but I cannot wrap my head around it.
I want to show a comparison between two objects with the same properties, 3 in total.

I want to show these properties next to each other in the bar chart, but I cannot wrap my head around the concept. As far as I can tell from the examples, the graph expects the same number of y-values as there are x-values. Currently I get the following result:

[screenshot of my current  graph](http://gyazo.com/1a1665e40e048e755dc35279fe283c4e)

my code looks like this:

```
ArrayList<String> xVals = new ArrayList<>();
xVals.add("Prop1");
xVals.add("Prop2");
xVals.add("Prop3");

ArrayList<BarEntry> yValsProp1 = new ArrayList<>();
ArrayList<BarEntry> yValsProp2 = new ArrayList<>();
ArrayList<BarEntry> yValsProp3 = new ArrayList<>();

yValsProp1.add(new BarEntry(dayOne.prop1, 0));
yValsProp1.add(new BarEntry(dayTwo.prop1, 1));

yValsProp2.add(new BarEntry(dayOne.prop2, 0));
yValsProp2.add(new BarEntry(dayTwo.prop2, 1));

yValsProp3.add(new BarEntry(dayOne.prop3, 0));
yValsProp3.add(new BarEntry(dayTwo.prop3, 1));

// create 3 datasets with different types
BarDataSet set1 = new BarDataSet(yValsProp1, "Prop1");
set1.setColor(getResources().getColor(R.color.purple));

BarDataSet set2 = new BarDataSet(yValsProp2, "Prop2");
set2.setColor(Color.rgb(164, 228, 251));

BarDataSet set3 = new BarDataSet(yValsProp3, "Prop");
set3.setColor(getResources().getColor(android.R.color.darker_gray));

ArrayList<BarDataSet> dataSets = new ArrayList<>();
dataSets.add(set1);
dataSets.add(set2);
dataSets.add(set3);

BarData data = new BarData(xVals, dataSets);

// add space between the dataset groups in percent of bar-width
data.setGroupSpace(80f);

barChart.setData(data);
barChart.invalidate();
```

Basically, i would like two purple bars next to each other, two blue, and two grey (the same properties next to eachother), how do I do this?

**Comment by PhilJay** [comment]
You are talking about a stacked-barchart: https://github.com/PhilJay/MPAndroidChart/blob/master/MPChartExample/src/com/xxmassdeveloper/mpchartexample/StackedBarActivity.java

Therefore, you need only one `DataSet`, containing 3 entries, each with 2 values. This is how you create an entry with 2 values:

`BarEntry e = new BarEntry(new float[] { 10f, 15f }, xIndex);`

**Comment by lfdversluis** [comment]
Hi thanks for your answer.

Is there a way to place the bars next to each other rather than stack them on top of each other?

**Comment by PhilJay** [comment]
Yes of course, just put your values in different entries.

**Comment by lfdversluis** [comment]
If I try add them in separate entries I get  ``java.lang.IllegalArgumentException: One or more of the DataSet Entry arrays are longer than the x-values array of this ChartData object.`` which makes sense. 

What I am trying to do is as follows: I have 3 sections in my graph (3 x-values). And I want to assign 2 entries per x-value (I want to give a comparison per section, think of calories eaten yesterday vs today as an example). Now If I wrap them like this: 

```
 yVals.add(new BarEntry(new float[] {
                dayOne.property1, dayTwo.property1
        }, 0));
```

They become stacked. How can I use entries with 2 values per section (x-value) and place the bars next to each other?

**Comment by PhilJay** [comment]
If you want 2 individual bars per x value, place the values in different datasets. If you want 2 individual values per entry, stack them like I showed you.

**Comment by lfdversluis** [comment]
Ah ok, so the combination cannot happen.
Thanks for your help.

**Comment by PhilJay** [comment]
Sure it can :-D
Just do it. Multiple datasets, with entries each containing multiple values.

**Comment by virip** [comment]
I want to know if it's possible overlap bars instead of stack them
Please help me!
